web/src/message_fetch.ts

```diff
@@ -300,21 +367,28 @@ export function load_messages(opts, attempt = 1) {
             // attempt to validate the narrow is compatible with
             // spectators here; the server will return an error if
             // appropriate.
-            data.narrow = JSON.parse(data.narrow);
-            data.narrow.push(web_public_narrow);
-            data.narrow = JSON.stringify(data.narrow);
+            const narrow_schema = z.array(
+                z.object({
+                    operator: z.string(),
+                    operand: z.union([z.string(), z.number(), z.array(z.number())]),
+                    negated: z.optional(z.boolean()),
+                }),
+            );
+            const parsed_narrow_data = narrow_schema.parse(JSON.parse(data.narrow));
+            parsed_narrow_data.push(web_public_narrow);
+            data.narrow = JSON.stringify(parsed_narrow_data);
         }
     }
 
-    channel.get({
+    void channel.get({
         url: "/json/messages",
         data,
         success(data) {
             if (!$("#connection-error").hasClass("get-events-error")) {
                 ui_report.hide_error($("#connection-error"));
             }
-
-            get_messages_success(data, opts);
+            const parsed_data = response_schema.parse(data);
```

Comment:
Might be better to use `raw_data` and `data` instead of `data` and `parsed_data`.

- https://github.com/zulip/zulip/pull/30521

explains the reason.

Reply:
Done!